File: cahoot/src/com/cahoot/eclipse/popup/actions/CollaboratorShareAction.java
package com.cahoot.eclipse.popup.actions;

import org.eclipse.jface.action.IAction;
import org.eclipse.jface.viewers.ISelection;
import org.eclipse.jface.window.ApplicationWindow;
import org.eclipse.swt.widgets.Shell;
import org.eclipse.ui.IActionDelegate;
import org.eclipse.ui.IObjectActionDelegate;
import org.eclipse.ui.IWorkbenchPart;

import com.cahoot.eclipse.collab.CollaboratorSelectionDialog;

public class CollaboratorShareAction implements IObjectActionDelegate {

	private Shell shell;
	
	/**
	 * Constructor for Action1.
	 */
	public CollaboratorShareAction() {
		super();
	}

	/**
	 * @see IObjectActionDelegate#setActivePart(IAction, IWorkbenchPart)
	 */
	public void setActivePart(IAction action, IWorkbenchPart targetPart) {
		shell = targetPart.getSite().getShell();
	}

	/**
	 * @see IActionDelegate#run(IAction)
	 */
	public void run(IAction action) {
		ApplicationWindow dialog = new CollaboratorSelectionDialog();
		dialog.open();
	}

	/**
	 * @see IActionDelegate#selectionChanged(IAction, ISelection)
	 */
	public void selectionChanged(IAction action, ISelection selection) {
	}

}

File: cahoot/src/main/scala/com/cahoot/eclipse/popup/actions/CollaboratorShareAction.scala
package com.cahoot.eclipse.popup.actions
import org.eclipse.ui.IObjectActionDelegate
import org.eclipse.swt.widgets.Shell
import org.eclipse.jface.action.IAction
import org.eclipse.ui.IWorkbenchPart
import org.eclipse.jface.window.ApplicationWindow
import org.eclipse.jface.viewers.ISelection
import com.cahoot.eclipse.collab.CollaboratorSelectionDialog

class CollaboratorShareAction extends IObjectActionDelegate {

  private var shell: Shell = _

  def setActivePart(action: IAction, targetPart: IWorkbenchPart): Unit = {
    shell = targetPart.getSite().getShell();
  }

  def run(action: IAction): Unit = {
    val dialog = new CollaboratorSelectionDialog();
    dialog.open();
  }

  /**
   * @see IActionDelegate#selectionChanged(IAction, ISelection)
   */
  def selectionChanged(action: IAction, selection: ISelection) = {
  }

}